addons/im_livechat/tests/test_member_history.py

```diff
@@ -0,0 +1,65 @@
+from odoo.addons.im_livechat.tests import chatbot_common
+from odoo.tests.common import tagged
+from odoo.addons.im_livechat.tests.common import TestGetOperatorCommon
+
+
+@tagged("post_install", "-at_install")
+class TestUserLivechatUsername(TestGetOperatorCommon, chatbot_common.ChatbotCase):
+    def test_get_session_create_history(self):
+        john = self._create_operator("fr_FR")
+        bob = self._create_operator("fr_FR")
+        livechat_channel = self.env["im_livechat.channel"].create(
+            {
+                "name": "Livechat Channel",
+                "user_ids": [bob.id],
+            }
+        )
+        data = self.make_jsonrpc_request(
+            "/im_livechat/get_session",
+            {
+                "anonymous_name": "Visitor",
+                "channel_id": livechat_channel.id,
+                "persisted": True,
+            },
+        )
+        channel = self.env["discuss.channel"].browse(data["channel_id"])
+        self.assertEqual(len(channel.channel_member_ids.channel_member_history_ids), 2)
+        self.assertEqual(
+            channel.channel_member_ids.channel_member_history_ids.filtered(
+                lambda m: m.partner_id == bob.partner_id
+            ).type,
+            "operator",
+        )
+        self.assertEqual(
+            channel.channel_member_ids.channel_member_history_ids.filtered(
+                lambda m: m.partner_id != bob.partner_id
+            ).type,
+            "visitor",
+        )
+        channel.add_members(partner_ids=john.partner_id.ids)
+        self.assertEqual(len(channel.channel_member_ids.channel_member_history_ids), 3)
```

Comment:
What's the point of adding a new member if we don't check it's type afterwards?

Reply:
This is just checking that he history is updated by add_members.
I can also add a test on the type, that is not really an issue.